types/three/examples/jsm/nodes/core/Node.d.ts

```diff
@@ -0,0 +1,29 @@
+import { NodeTypeOption, NodeUpdateTypeOption } from './constants';
+import NodeBuilder from './NodeBuilder';
+import NodeFrame from './NodeFrame';
+
+export default class Node {
+    uuid: string;
+    type: string;
+    isNode: true;
+    nodeType: NodeTypeOption;
+    updateType: NodeUpdateTypeOption;
+    id: number;
+
+    constructor(nodeType?: NodeTypeOption);
+
+    getChildren(): Node[];
+    getHash(builder: NodeBuilder): string;
+    getUpdateType(builder: NodeBuilder): NodeUpdateTypeOption;
+    getNodeType(builder: NodeBuilder, output?: string | null): NodeTypeOption | null;
```

Comment:
IMO this isn't correct for any of the `builder` stuff, in this class specifically it's not there...

Reply:
The subclasses require it. Not including it would break the inheritance. You cannot really override the method in subclass and add a parameter

(what nodes do in javascript is also correct, as including the parameter wouldn't change anything, javascript will happily call a method without parameter)